src/mrmailbox_imex.c

```diff
@@ -815,7 +814,17 @@ static int import_self_keys(mrmailbox_t* mailbox, const char* dir_name)
 		buf2 = safe_strdup(buf);
 		if( mr_split_armored_data(buf2, &buf2_headerline, NULL, NULL, NULL)
 		 && strcmp(buf2_headerline, "-----BEGIN PGP PUBLIC KEY BLOCK-----")==0 ) {
-			continue; /* this is no error but quite normal as we always export the public keys together with the private ones */
+			/* This file starts with a Public Key.
+			 * However some programs (Thunderbird/Enigmail) put public and private key
+			 * in the same file, so we check if there is a private key following */
+			char* following_private_key = strstr(buf, "-----BEGIN PGP PRIVATE KEY BLOCK");
+			if (following_private_key == NULL) {
+				continue; /* this is no error but quite normal as we always export the public keys together with the private ones */
+			}
+			/*
+			 * If so, we point the buffer to that private key
+			 */
+			buf = following_private_key;
```

Comment:
this will cause problems when free() is called for `buf` afterwards (`following_private_key` is just a pointer inside `buf` and is not a pointer returned by malloc() and cannot be free()'d).

the solution may be to copy buf using memcpy() or to use an additional const-pointer key_start or so that is set up but not free()'d.

maybe we can use a generic `const char* private_key` that is equal to buf by default but may be modified and is never free()'d.